src/database/DataTypes/TemperatureBreachConfiguration.js

```diff
@@ -4,8 +4,69 @@
  */
 
 import Realm from 'realm';
+import moment from 'moment';
+import { createRecord } from '../utilities';
 
-export class TemperatureBreachConfiguration extends Realm.Object {}
+export class TemperatureBreachConfiguration extends Realm.Object {
+  /**
+   * With the provided temperatureLog, determines if the passed location
+   * was in a breach at the time of the temperature log and if so, will
+   * create a temperature breach record. Also adds to each of the temperature
+   * logs which are currently part of that breach, the related breach
+   * record.
+   *
+   * A breach has a duration and minimum and maximum temperatures defined by this
+   * configuration - for example above 8 degrees for 2 hours. To determine if a
+   * location has currently been breached, look back from the temperature logs
+   * timestamp the duration of this configuration and determine if every temperature
+   * log since that time has been outside of the threshold.
+   *
+   * @param {Realm}          database App-wide database interface
+   * @param {Location}       location A Location record to create a breach for.
+   * @param {TemperatureLog} temperatureLog A potential log which may create a breach.
+   */
+  createBreach(database, location, temperatureLog) {
```

Comment:
I like the suggestion of splitting up this method. Like you say, it is actually fine like it is - I think it would be slightly neater if split into the three methods.

I'd filter the logs in `createBreach` and pass the `proximalLogs` down rather than fetching twice